CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -804,9 +810,198 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
+
+                if (playerBets.All(n => n == playerBets[0])) //if all bets match the ante
+                {
+                    AddToPot(_ante, playerBets, playersActive);
+                }
+                else //create side pots if any bets do not match the ante
+                {
+                    CreateSidePots(_ante, playerBets, playersActive);
+                }
+            }
+            /// <summary>
+            /// Adds a bets to the pot.
+            /// </summary>
+            /// <param name="_currentBet">The amount each player is contributing to the pot.</param>
+            /// <param name="numPlayers">The number of players contributing to the pot.</param>
+            /// <param name="playersActive">The positions of the players that are currently eligible for the pot</param>
+            public void AddToPot(int _currentBet, List<int> playerBets, List<int> playersActive)
+            {
+                if (playerBets.All(n => n == playerBets[0]))
+                {
+                    Pots.First(pot => pot.PotType == PotType.MAIN).Total += _currentBet * playerBets.Count();
+                }
+                else
+                {
+                    CreateSidePots(_currentBet, playerBets, playersActive);
+                }
+            }
+            /// <summary>
+            //Adds bets of players that were folded to the pot
+            /// </summary>
+            /// <param name="playersBets"> list of bets that were placed but folded due to a raise in that round</param>
+            public void AddFoldedBetsToPot(List<int> playerBets)
+            {
+                if (playerBets.Count() == 0)
+                {
+                    return;
+                }
+                else
+                {
+                    for (int bet = 0; bet < playerBets.Count(); bet++)
+                    {
+                        Pots[0].Total += playerBets[bet];
+                    }
+                }
+            }
+            /// <summary>
+            /// Creates side pots as needed at the end of a round of betting.
+            /// </summary>
+            /// <param name="_currentBet">The amount to add to the pot.</param>
+            /// <param name="playerBets">The list of bets from players still active after the round of betting.</param>
+            /// <param name="playersActive">The list of players eligible to win the main pot, all in players will be subtracted for side pot</param>
+            public void CreateSidePots(int _currentBet, List<int> playerBets, List<int> playersActive)
+            {
+                int allInBet = 0; //initiating variable that will hold all in wager
+                int numBets = playerBets.Count();
+
+                for (int bets = 0; bets < numBets; bets++)
+                {
+                    //at least 1 side pot is needed if function is called, lowest bet will be an all-in bet.
+                    if (bets == 0)
```

Comment:
instead of nesting this as an if clause within the for loop, lets just call all of this logic once before the for loop and unnest it a bit.

Reply:
Removed the for loop as requested. Let me know if there are any additional changes I can implement for it!